src/chatkit.ts

```diff
@@ -11,17 +11,26 @@ import {
   TokenWithExpiry,
 } from "@pusher/platform-node"
 
-import { getCurrentTimeInSeconds } from "./utils"
+import {
+  getCurrentTimeInSeconds,
+  validateProperties,
+} from "./utils"
 import packageJSON from "../package.json"
+import * as t from 'io-ts'
+import "unknown-ts"
+
+const NonEmptyString = t.refinement(t.string, s => s.length != 0, "NonEmptyString");
 
 export interface AuthenticationOptions {
   userId: string
   authPayload?: AuthenticatePayload
 }
 
-export interface UserIdOptions {
-  userId: string
-}
+const UserIdOptions = t.interface({
+  userId: NonEmptyString
+},"UserIdOptions");
+type UserIdOptions = t.TypeOf<typeof UserIdOptions>;
+export { UserIdOptions };
```

Comment:
The use of the same name for two different things is confusing at a glance (though I know you mentioned the docs gave an example like this)